Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp

```diff
@@ -499,4 +505,73 @@ void FiniteDifferenceSolver::EvolveBCylindrical (
     }
 }
 
-#endif // corresponds to ifndef WARPX_DIM_RZ
+#elif defined(WARPX_DIM_RSPHERE)
+
+template<typename T_Algo>
+void FiniteDifferenceSolver::EvolveBSpherical (
+    ablastr::fields::VectorField const& Bfield,
+    ablastr::fields::VectorField const& Efield,
+    int lev, amrex::Real const dt ) {
+
+    amrex::LayoutData<amrex::Real>* cost = WarpX::getCosts(lev);
+
+    // Loop through the grids, and over the tiles within each grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for ( MFIter mfi(*Bfield[0], TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            amrex::Gpu::synchronize();
+        }
+        auto wt = static_cast<amrex::Real>(amrex::second());
+
+        // Extract field data for this grid/tile
+        Array4<Real> const& Br = Bfield[0]->array(mfi);
+        Array4<Real> const& Bt = Bfield[1]->array(mfi);
+        Array4<Real> const& Bp = Bfield[2]->array(mfi);
+        Array4<Real> const& Et = Efield[1]->array(mfi);
+        Array4<Real> const& Ep = Efield[2]->array(mfi);
+
+        // Extract stencil coefficients
+        Real const * const AMREX_RESTRICT coefs_r = m_stencil_coefs_r.dataPtr();
+        auto const n_coefs_r = static_cast<int>(m_stencil_coefs_r.size());
+
+        // Extract spheriical specific parameters
+        Real const dr = m_dr;
+        Real const rmin = m_rmin;
+
+        // Extract tileboxes for which to loop
+        Box const& tbr  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box const& tbt  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box const& tbp  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // Loop over the cells and update the fields
+        amrex::ParallelFor(tbr, tbt, tbp,
+
+            [=] AMREX_GPU_DEVICE (int i, int /*j*/, int /*k*/){
+                Br(i, 0, 0, 0) = 0.;
+            },
+
+            [=] AMREX_GPU_DEVICE (int i, int /*j*/, int /*k*/){
+                Real const r = rmin + (i + 0.5_rt)*dr; // r on a cell-centered grid (Bp is cell-centered in r)
+                Bt(i, 0, 0, 0) += dt*( + T_Algo::UpwardDrr_over_r(Ep, r, dr, coefs_r, n_coefs_r, i, 0, 0, 0));
+            },
+
+            [=] AMREX_GPU_DEVICE (int i, int /*j*/, int /*k*/){
+                Real const r = rmin + (i + 0.5_rt)*dr; // r on a cell-centered grid (Bt is cell-centered in r)
+                Bp(i, 0, 0, 0) += dt*( - T_Algo::UpwardDrr_over_r(Et, r, dr, coefs_r, n_coefs_r, i, 0, 0, 0));
```

Comment:
The PSATD RZ solver is an example where `Bp` is used to mean something different, namely $B^+$ instead of $B^\phi$ in that case. Regardless of whether in this PR or in a follow-up PR, I would personally be in favor of not using the same naming convention to mean different things, unless there is a clear benefit. For example, we could spell out both variables as `Bplus` and `Bphi` or `B_plus` and `B_phi` (the latter with the understanding that `_` does not mean "subscript" in the covariant vs contravariant sense).

Reply:
These were changed to Bphi etc